src/utils.ts

```diff
@@ -62,7 +62,7 @@ export const createCache = (): FastCopy.Cache => {
  * @returns the empty cloned object
  */
 export const getCleanClone = (object: any, realm: FastCopy.Realm): any => {
-  if (!object.constructor) {
+  if (!object.constructor || !(object.constructor instanceof Function)) {
```

Comment:
Two things:

While I think this will work around the problem, it is not a true "copy". In your test example, it is copying a standard object, which would have a prototype of `Object`, but the copy would have a prototype of `null`, meaning all the standard prototype methods would not be inherited. This is also true of custom prototype objects, because JS can be super flexible:

```js
const instance = new CustomThing();

instance.constructor = 'hello';
```

I would instead recommend gating the `toString` check on the fact that `constructor` is a function:

```ts
export const getCleanClone = (object: any, realm: FastCopy.Realm): any => {
  if (!object.constructor) {
    return create(null);
  }

  const { constructor: Constructor } = object;
  const prototype = object.__proto__ || getPrototypeOf(object);

  if (Constructor === realm.Object) {
    return prototype === realm.Object.prototype ? {} : create(prototype);
  }

  if (typeof Constructor === 'function' && ~toStringFunction.call(Constructor).indexOf('[native code]')) {
    try {
      return new Constructor();
    } catch {}
  }

  return create(prototype);
};
```

This would eliminate the error, but also ensure the custom object was created with the correct prototype.

Reply:
One additional thing: noticed I used `typeof` instead of `instanceof`. In modern engines the latter is quite fast, however in older engines it is significantly slower. For this purpose there would be no result differences between the two, so we should default to the one that is fast in all engines.